packages/vm/src/runTx.ts

```diff
@@ -440,6 +447,53 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
     fromAccount.balance = BIGINT_0
   }
   await this.evm.journal.putAccount(caller, fromAccount)
+
+  const writtenAddresses = new Set<string>()
+  if (tx.supports(Capability.EIP7702EOACode)) {
+    // Add contract code for authroity tuples provided by EIP 7702 tx
+    const authorizationList = (<EIP7702CompatibleTx>tx).authorizationList
+    const MAGIC = new Uint8Array([5])
+    for (let i = 0; i < authorizationList.length; i++) {
+      // Authority tuple validation
+      const data = authorizationList[i]
+      const chainId = data[0]
+      const chainIdBN = bytesToBigInt(chainId)
+      if (chainIdBN !== BIGINT_0 && chainIdBN !== this.common.chainId()) {
+        // Chain id does not match, continue
+        continue
```

Comment:
Ok, just seeing this here.

So I would interpret this (still without reading the EIP): txs with non-matching chain IDs are valid (could be included in mainnet), but execution just "does nothing" if not matching.

Is this correct/intended?

Reply:
Yes, from the EIP:

`If any of the above steps fail, immediately stop processing that tuple and continue to the next tuple in the list. `

The list contains point (2) the chain id check, it should either be 0 or the current chain id. If it thus does not match (and it is nonzero) then we continue processing, but it thus does not change any code. (Point 7 is changing the code of the authority).

See: https://eips.ethereum.org/EIPS/eip-7702#behavior